client/hostvolumemanager/host_volumes.go

```diff
@@ -26,42 +27,68 @@ type HostVolumeStateManager interface {
 	DeleteDynamicHostVolume(string) error
 }
 
+type Config struct {
+	// PluginDir is where external plugins may be found.
+	PluginDir string
+
+	// SharedMountDir is where plugins should place the directory
+	// that will later become a volume HostPath
+	SharedMountDir string
+
+	// StateMgr manages client state to restore on agent restarts.
+	StateMgr HostVolumeStateManager
+
+	// UpdateNodeVols is run to update the node when a volume is created
+	// or deleted.
+	UpdateNodeVols HostVolumeNodeUpdater
+}
+
 type HostVolumeManager struct {
 	pluginDir      string
 	sharedMountDir string
 	stateMgr       HostVolumeStateManager
-
-	log hclog.Logger
+	updateNodeVols HostVolumeNodeUpdater
+	log            hclog.Logger
 }
 
-func NewHostVolumeManager(logger hclog.Logger,
-	state HostVolumeStateManager, restoreTimeout time.Duration,
-	pluginDir, sharedMountDir string) (*HostVolumeManager, error) {
-
-	log := logger.Named("host_volume_mgr")
-
-	// db TODO(1.10.0): how do we define the external mounter plugins? plugin configs?
-	hvm := &HostVolumeManager{
-		pluginDir:      pluginDir,
-		sharedMountDir: sharedMountDir,
-		stateMgr:       state,
-		log:            log,
+func NewHostVolumeManager(logger hclog.Logger, config Config) *HostVolumeManager {
+	// db TODO(1.10.0): document plugin config options
+	return &HostVolumeManager{
+		pluginDir:      config.PluginDir,
+		sharedMountDir: config.SharedMountDir,
+		stateMgr:       config.StateMgr,
+		updateNodeVols: config.UpdateNodeVols,
+		log:            logger.Named("host_volume_manager"),
 	}
+}
 
-	if err := hvm.restoreState(state, restoreTimeout); err != nil {
-		return nil, err
+func genVolConfig(req *cstructs.ClientHostVolumeCreateRequest, resp *HostVolumePluginCreateResponse) *structs.ClientHostVolumeConfig {
+	if req == nil || resp == nil {
+		return nil
+	}
+	return &structs.ClientHostVolumeConfig{
+		Name: req.Name,
+		ID:   req.ID,
+		Path: resp.Path,
+
+		// dynamic volumes, like CSI, have more robust `capabilities`,
+		// so we always set ReadOnly to false, and let the scheduler
+		// decide when to ignore this and check capabilities instead.
```

Comment:
I think the taskrunner's `volume_hook` will need to handle this too, but I'll follow-up with that later.